config-model/src/main/java/com/yahoo/vespa/model/content/cluster/ContentCluster.java

```diff
@@ -430,8 +430,11 @@ private ClusterControllerContainerCluster createClusterControllers(AbstractConfi
             if (clusterControllers.getContainers().isEmpty()) {
                 int index = 0;
                 for (HostResource host : hosts) {
+                    int ccIndex = deployState.getProperties().dedicatedClusterControllerCluster()
+                                  ? host.spec().membership().map(ClusterMembership::index).orElse(index)
+                                  : index;
                     boolean retired = host.spec().membership().map(ClusterMembership::retired).orElse(false);
-                    var clusterControllerContainer = new ClusterControllerContainer(clusterControllers, index, multitenant, deployState, retired);
+                    var clusterControllerContainer = new ClusterControllerContainer(clusterControllers, ccIndex, multitenant, deployState, retired);
```

Comment:
Not 100% sure on the impact of this. Does it affect CC leader election?

Reply:
The CC index is set below, in `addComponentsForThisCluster`. I'm also not certain this will work, so will keep an eye on CD.